### ISSUE TITLE
Be more specific about rust version

### ISSUE THREAD

**Issue by NobbZ** [indiv0/colonize#49]
I tried to compile and run `colonize` from master a couple of minuets ago, but it does not work, I do get different error messages depending on the toolchain used to compile, I will give an excerpt below:

But lets start with the toolchain versions I tested:

```
$ multirust run nightly rustc --version
rustc 1.10.0-nightly (4ec5ce5e4 2016-05-12)

$ multirust run beta rustc --version
rustc 1.9.0-beta.2 (87cb73304 2016-05-04)

$ multirust run stable rustc --version
rustc 1.8.0 (db2939409 2016-04-11)
```

Now the failure I do get when trying to compile stable or beta:

```
   Compiling colonize_world v0.0.1 (file:///home/NobbZ/projects/rust/colonize/world)
   Compiling rgframework v0.0.1 (file:///home/NobbZ/projects/rust/colonize/framework)
   Compiling colonize v0.0.2 (file:///home/NobbZ/projects/rust/colonize)
src/action.rs:3:30: 3:39 error: `#[derive]` for custom traits is not stable enough for use and is subject to change (see issue #29644)
src/action.rs:3 #[derive(Clone, Deserialize, Serialize)]
                                             ^~~~~~~~~
[... similar messages ...]
src/localization.rs:5:1: 5:56 error: couldn't read "/home/NobbZ/projects/rust/colonize/target/debug/build/colonize-edfe5b092dfa28d2/out/localization.rs": No such file or directory (os error 2)
src/localization.rs:5 include!(concat!(env!("OUT_DIR"), "/localization.rs"));
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: Could not compile `colonize`.
```

and nightly does emit the following stuff:

```
   Compiling rustc-serialize v0.3.19
   Compiling deque v0.3.1
   Compiling rayon v0.3.1
/home/NobbZ/.multirust/cargo/registry/src/github.com-1ecc6299db9ec823/rayon-0.3.1/src/par_iter/enumerate.rs:21:5: 25:6 error: the trait bound `C: par_iter::internal::Consumer<par_iter::enumerate::Enumerate<M>>` is not satisfied [E0277]
/home/NobbZ/.multirust/cargo/registry/src/github.com-1ecc6299db9ec823/rayon-0.3.1/src/par_iter/enumerate.rs:21     fn drive_unindexed<C>(self, consumer: C) -> C::Result
                                                                                                                     ^
[... many other messages tagged [E0277] ...]
error: aborting due to 55 previous errors
Build failed, waiting for other jobs to finish...
^C
```

The very same error appears never the less when I use the command as specified in the README or a plain `cargo run`.

**Comment by indiv0** [comment]
Very interesting.

For me it compiles on both nightly:

```
─── multirust run nightly rustc --version
rustc 1.10.0-nightly (2b79e05a0 2016-05-13)
─── multirust run nightly cargo build --no-default-features --features nightly 
```

and stable:

```
─── multirust run stable rustc --version
rustc 1.8.0 (db2939409 2016-04-11)
─── multirust run stable cargo build
```

I'm not sure about beta, but it should be compiling on stable and nightly for you.
Can you post the exact `cargo build`/`cargo run` commands you used?

**Comment by indiv0** [comment]
Actually, regarding your nightly build. That's an [issue with `rayon`](https://github.com/nikomatsakis/rayon/issues/77) that was specific to the `2016-05-12` nightly so if you update to `2016-05-13` it should compile.

As to your stable and beta builds, it still looks to me like you used the wrong build commands.

**Comment by indiv0** [comment]
That said, the project could definitely use more information regarding which rustc versions are compatible. That should be fixed when I fix #3 and #7 by adding `travis-cargo` to the `.travis.yml` and when I add the relevant information to the `README.md` when I fix #13.

**Comment by indiv0** [comment]
Actually testing the build on stable results in the same errors that you got. My fault for assuming it worked (it did work just a few days ago!).

Looking into it now.

**Comment by NobbZ** [comment]
Commands for building/running where simple as cargo run/build...

indiv0 <notifications@github.com> schrieb am Sa., 14. Mai 2016, 22:27:

**Comment by indiv0** [comment]
And on nightly?

**Comment by NobbZ** [comment]
Plain as on stable/beta and copy pasted the ones from the README.

indiv0 <notifications@github.com> schrieb am Sa., 14. Mai 2016, 22:37:

**Comment by indiv0** [comment]
OK. I've found the issue and I'm working to correct it now.

At the moment, if you update to the latest nightly you should be able to compile with `cargo build --no-default-features --features nightly`, but compilation on stable and beta is broken.

**Comment by NobbZ** [comment]
After I had to purge and reinstall rustic because of former
selfdesintegration of multitude I'm already trying on latest nightly. It's
only a couple of minutes now...

indiv0 <notifications@github.com> schrieb am Sa., 14. Mai 2016, 22:49:

**Comment by indiv0** [comment]
@NobbZ https://github.com/indiv0/colonize/pull/51 should fix compilation on stable and beta. Try it and let me know.

**Comment by NobbZ** [comment]
Ok, I checked commit 3e88676 from master since you already merged #51, and I can build from all relevant toolchains now.

Thank you!

**Comment by indiv0** [comment]
You're welcome! Let me know if you have any further questions or issues.
